pkg/reconciler/kubernetes/tektonconfig/controller.go

```diff
@@ -79,6 +82,19 @@ func NewExtendedController(generator common.ExtensionGenerator) injection.Contro
 			Handler:    controller.HandleAll(impl.EnqueueControllerOf),
 		})
 
+		createInstance := os.Getenv("TEKTON_CONFIG_INSTANCE")
+		if createInstance == "true" {
+			tci := tektonConfigInstance{
+				operatorClientSet: operatorclient.Get(ctx),
+				kubeClientSet:     kubeclient.Get(ctx),
+				manifest:          manifest,
+				namespace:         defaultNamespace,
+			}
+
+			// try to ensure that there is an instance of tektonConfig
+			tci.ensureInstance(ctx)
```

Comment:
could you move this into a function, for example `NewTektonConfig` and pass all these as arguments.
it will be more readable.